starknet_py/compile/compiler.py

```diff
@@ -27,7 +31,7 @@ class Compiler:
 
     def __init__(
         self,
-        contract_source: List[StarknetCompilationSource],
+        contract_source: StarknetCompilationSource,
         cairo_path: Optional[List[str]] = None,
     ):
```

Comment:
Also since we're accepting cairo path as list of strings, I've added `str()` cast to places where we used `Path`. Perhaps we should use `Path` here instead?

Reply:
You mean we should require Path? It is not ergonomic

Reply:
That's what i meant. I'm fine with still accepting string and adding necessary casts though (also it works without explicit casting so it doesn't matter for end users)

Reply:
If Path is required by cairo-lang then can't we just accept strings and map them to Path, removing casts?

Reply:
It's the other way around, we're internally passing Paths multiple times, when actually there not used anywhere in cairo-lang or our compiler module

Reply:
Hmmm do we lose anything if we switched to strings?

Reply:
Probably the convenience that comes with using `Path` instead of raw strings. [See here for example of casts I added.](https://github.com/software-mansion/starknet.py/pull/185/files#r889915693)

Reply:
Where do we use utilities from Path?

Reply:
ok 👍